WordPress/gutenberg
# Font Family Picker: allow selecting the "Default" font for all text elements and blocks, which didn't work with some themes

**Not ready yet - more to follow.** 



## What?
- Allow users to select the "Default" font on themes that specify `fontFamily` at [element](https://github.com/Automattic/themes/blob/trunk/pendant/theme.json#L302) or block-level, e.g. Pendant. 
- Add a hint to the Heading Typography panel to remind users to set Post Title and Site Title styles in the Blocks menu (this is not obvious and caught me out).
- Update `TypographyPanel` so it reflects the Editor & Site `font-family`
- This PR also links heading elements and heading blocks. Now, if you set your Site Title font to "Default", it will update whenever you change the heading element, e.g. H1, that it is based on. 

## Why?
- If you wanted to set your All Heading Font to "X" and then set, say, your H2 to "Default", then you would expect the H2 `font-family` to use the "X" font, but this didn't work on some themes like Pendant. 
- It's not obvious when you're editing a Site Title vs a regular heading element

## How?
The "Default" font value was `''` which is removed by the [config merge logic](https://github.com/WordPress/gutenberg/blob/58c78ec5200a38401281c14186095bced2f6f12a/packages/edit-site/src/components/global-styles/global-styles-provider.js#L88) and meant the [base value](https://github.com/Automattic/themes/blob/trunk/pendant/theme.json#L309) was always used if it was present. This PR allows empty values to pass through and means the "Default" font style can be applied from a higher level. 

**Note:** there is a question about using this approach: was the use of `cleanEmptyObject` intended to stop unnecessary network traffic, or was it for some other reason? An empty value is valid and necessary according to my tests with various Global Style controls, but maybe I missed something. Please check this as part of the review. 


## Testing Instructions

1. Using the Pendant theme, try to change a heading style, e.g. H2, to "Default" in the **Global Styles** > **Typography** > **Headings** menu.
2. Note how it always defaults to "Headings (System Font)".  You can try different themes to compare. Disco, for example, does not have this problem because it doesn't specify `fontFamily` at the element-level in `themes.json`. 
3. Now checkout this branch and start your local dev environment
4. You should now be able to change heading fonts to "Default, " meaning they use the "All" heading font. If you also set a heading block, like Site Title, to "Default", it will inherit the font from the regular H2 block, which means you can now manage all of your headings via the **Global Styles** > **Typography** > **Headings** menu.
5. Check any combination of typography styles you can think of and verify the `TypographyPanel` preview, Editor and Site reflect the styles correctly. I tested with Blockbase, Pendant and Twenty Twenty-Two.
6. Check for regressions. This PR relaxes a check for empty values, which will now show as unsaved changes and be sent to the server. 


## Screenshots or screencast 

The new hint:
![image](https://user-images.githubusercontent.com/6851384/191663275-ceb3e809-fbd3-49c8-946b-79c3b36c16f6.png)

Example of the problem (from WordPress.com):
...
